motr/iem.c

```diff
@@ -79,22 +79,41 @@ void m0_iem(const char* file, const char* function, int line,
 		vsnprintf(description, sizeof(description)-1, msg, aptr);
 		va_end(aptr);
 	}
-	if (++iem_re[evt_id].ir_niems < THRESHOLD) {
-		m0_console_printf("IEC: %c%c%03x%03x%04x: counts %"PRIx64", "
-				  "%s\n",
+	if (mod_id == M0_MOTR_IEM_MODULE_IO &&
+	    evt_id == M0_MOTR_IEM_EVENT_MD_ERROR) {
+		if (++iem_re[evt_id].ir_niems < THRESHOLD)
+			m0_console_printf("IEC: %c%c%03x%03x%04x: "
+					  "IO errors (%"PRIu64") encountered. "
+					  "%s\n",
+					  m0_motr_iem_severity[sev_id],
+					  M0_MOTR_IEM_SOURCE_ID,
+					  M0_MOTR_IEM_COMPONENT_ID_MOTR, mod_id,
+					  evt_id, iem_re[evt_id].ir_total,
+					  description);
+	} else
+		m0_console_printf("IEC: %c%c%03x%03x%04x: %s\n",
 				  m0_motr_iem_severity[sev_id],
 				  M0_MOTR_IEM_SOURCE_ID,
 				  M0_MOTR_IEM_COMPONENT_ID_MOTR,
-				  mod_id, evt_id, iem_re[evt_id].ir_total,
-				  description);
-		m0_console_flush();
-	}
+				  mod_id, evt_id, description);
+
+	m0_console_flush();
+
 	/* Do not throttle trace messages. */
-	M0_LOG(M0_INFO, "IEC: %c%c%3x%3x%4x: counts %"PRIx64", %s",
-	       m0_motr_iem_severity[sev_id],
-	       M0_MOTR_IEM_SOURCE_ID, M0_MOTR_IEM_COMPONENT_ID_MOTR,
-	       mod_id, evt_id, iem_re[evt_id].ir_total,
-	       (const char*)description);
+	if (mod_id == M0_MOTR_IEM_MODULE_IO &&
+	    evt_id == M0_MOTR_IEM_EVENT_MD_ERROR)
+		M0_LOG(M0_INFO, "IEC: %c%c%3x%3x%4x: "
+		       "IO errors (%"PRIx64")encountred. %s",
+		       m0_motr_iem_severity[sev_id],
+		       M0_MOTR_IEM_SOURCE_ID, M0_MOTR_IEM_COMPONENT_ID_MOTR,
+		       mod_id, evt_id, iem_re[evt_id].ir_total,
+		       (const char *)description);
+	else
+		M0_LOG(M0_INFO, "IEC: %c%c%3x%3x%4x: %s",
+		       m0_motr_iem_severity[sev_id],
+		       M0_MOTR_IEM_SOURCE_ID, M0_MOTR_IEM_COMPONENT_ID_MOTR,
+		       mod_id, evt_id, (const char *)description);
+
 	M0_LOG(M0_INFO, "from %s:%s:%d",
 	       (const char*)file, (const char*)function, line);
```

Comment:
Do we still need this M0_LOG()?

Reply:
It has been from before and would be needed since the throttled IEM messages would still be logged.